index.js

```diff
@@ -28,39 +23,51 @@ function basicPlugin (fastify, opts, next) {
       }
     }
 
-    function done (err) {
-      if (err !== undefined) {
+    function done (err, realm) {
+      // TODO remove in the next major
+      if (typeof err === 'string') {
+        realm = err
+        err = undefined
+      }
+      if (err) {
         // We set the status code to be 401 if it is not set
         if (!err.statusCode) {
           err.statusCode = 401
         }
         next(err)
       } else {
+        const header = realm ? formatRealm(realm) : authenticateHeader
+        reply.header('WWW-Authenticate', header)
```

Comment:
Should not this header be added on 401 only?

https://developer.mozilla.org/en-US/docs/Web/HTTP/Authentication

Reply:
You are right, I made a mess. I'll revert and restore the double callback. Sigh.